### PR TITLE
WT-8286 Create stress test for prefix search

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: test/cppsuite/configs/search_near_01_stress.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+(
+    populate_config=
+    (
+        collection_count=3,
+        key_count_per_collection=100,
+        key_size=5,
+    ),
+    read_config=
+    (
+        thread_count=10
```
Comment (etienneptl) [inline test/cppsuite/configs/search_near_01_stress.txt:20]: @jiechenbo, should we increase the `op_rate` for the `read_operation` threads?

File: test/cppsuite/configs/search_near_01_stress.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+(
+    enabled=false,
+),
+workload_generator=
+(
+    populate_config=
+    (
+        collection_count=3,
+        key_count_per_collection=100,
+        key_size=5,
```
Comment (etienneptl) [inline test/cppsuite/configs/search_near_01_stress.txt:16]: We could potentially increase the `key_size` to force the keys to be split across multiple pages?

File: test/evergreen.yml [hunk 3]
```diff
@@ -3795,6 +3810,7 @@ buildvariants:
     - name: compile
     - name: cppsuite-hs-cleanup-stress
```
Comment (etienneptl) [inline test/evergreen.yml:3811]: Sorry, it's not you but can we alphabetize those tests?

File: test/cppsuite/configs/search_near_01_stress.txt (hunk 2 above)
Comment (quchenhao) [reply test/cppsuite/configs/search_near_01_stress.txt:16]: I increases key_count_per_collection to achieve that. I think that is more reliable to reproduce that situation. Of course we can increase both.

File: test/cppsuite/configs/search_near_01_stress.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+timestamp_manager=
+(
+    enabled=false,
+),
+workload_generator=
+(
+    populate_config=
+    (
+        collection_count=3,
+        key_count_per_collection=100,
```
Comment (etienneptl) [inline test/cppsuite/configs/search_near_01_stress.txt:15]: @jiechenbo , I had a look at the code and this does not seem to be used at all. We should remove it from both cfgs?

File: test/cppsuite/configs/search_near_01_stress.txt (hunk 2 above)
Comment (etienneptl) [reply test/cppsuite/configs/search_near_01_stress.txt:16]: Please see my comment [here](https://github.com/wiredtiger/wiredtiger/pull/7151/files#r743363359). I am afraid we are not using this parameter, hence we will need to use the `key_size` field.

File: test/cppsuite/configs/search_near_01_stress.txt (hunk 4 above)
Comment (etienneptl) [reply test/cppsuite/configs/search_near_01_stress.txt:15]: If this setting is not used, we should remove it from the code.

File: test/cppsuite/configs/search_near_01_stress.txt (hunk 2 above)
Comment (etienneptl) [reply test/cppsuite/configs/search_near_01_stress.txt:16]: Sorry for my previous comments (I deleted them), I misread the code. I am happy to keep it to `100` if that's enough to get the keys to be on multiple pages. Could you verify it?

File: test/cppsuite/configs/search_near_01_stress.txt (hunk 1 above)
Comment (jiechenbo) [reply test/cppsuite/configs/search_near_01_stress.txt:20]: I realise we call `tc->sleep();` to sleep on the read operation. Does op_rate configure the amount sleep? If so we definitely should!

File: test/cppsuite/configs/search_near_01_stress.txt (hunk 2 above)
Comment (jiechenbo) [reply test/cppsuite/configs/search_near_01_stress.txt:16]: Maybe we could do both? Having a big key size and a lot of keys might be quite beneficial.

File: test/cppsuite/configs/search_near_02_stress.txt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+),
+workload_generator=
+(
+    populate_config=
+    (
+        collection_count=10,
+    ),
+    insert_config=
+    (
+        key_size=5,
```
Comment (jiechenbo) [inline test/cppsuite/configs/search_near_02_stress.txt:21]: Thinking we can add a big key size here too? Might add some benefit where key size are really big.

File: test/cppsuite/configs/search_near_01_stress.txt (hunk 2 above)
Comment (quchenhao) [reply test/cppsuite/configs/search_near_01_stress.txt:16]: I'm able to reproduce WT-7198 so I believe it is working.

File: test/cppsuite/configs/search_near_01_stress.txt (hunk 1 above)
Comment (quchenhao) [reply test/cppsuite/configs/search_near_01_stress.txt:20]: What should it be increased to?

Comment (etienneptl) [reply test/cppsuite/configs/search_near_01_stress.txt:20]: Like the other config, `10ms` should be good? At the end of the day, we need to check on T2 that the system is being stressed in the expected areas.

File: test/cppsuite/configs/search_near_02_stress.txt (hunk 5 above)
Comment (etienneptl) [reply test/cppsuite/configs/search_near_02_stress.txt:21]: I agree. Please @quchenhao, can you check you reproduce WT-7912 with this config too?

File: test/cppsuite/configs/search_near_01_stress.txt (hunk 1 above)
Comment (quchenhao) [reply test/cppsuite/configs/search_near_01_stress.txt:20]: @jiechenbo What do you mean? Larger value means a higher rate?

Comment (etienneptl) [reply test/cppsuite/configs/search_near_01_stress.txt:20]: Yes, this is what we mean, increase the rate of execution. The default value is `1s` as defined [here](https://github.com/wiredtiger/wiredtiger/blob/develop/dist/test_data.py#L47-L51), a smaller value for a higher rate. Sorry for the confusion.

Comment (quchenhao) [reply test/cppsuite/configs/search_near_01_stress.txt:20]: @etienneptl If smaller is a higher rate, then the rate is already increased.

File: test/cppsuite/configs/search_near_02_stress.txt (hunk 5 above)
Comment (quchenhao) [reply test/cppsuite/configs/search_near_02_stress.txt]: @etienneptl Just key size?

File: test/cppsuite/configs/search_near_01_stress.txt [hunk 6]
```diff
@@ -13,10 +13,14 @@ workload_generator=
...
         key_count_per_collection=100,
-        key_size=50,
+        key_size=5,
     ),
     read_config=
     (
+        # The key size parameter is used to generate the prefix 
+        # used in search_near. Use a smaller size to increase
+        # the chance to match an existing key.
+        key_size=3,
```
Comment (etienneptl) [inline test/cppsuite/configs/search_near_01_stress.txt:23]: @jiechenbo , would I be wrong if I say the `key_size` is not used in `search_near_01.cxx`?

Comment (quchenhao) [reply test/cppsuite/configs/search_near_01_stress.txt:23]: @jiechenbo Is the key_size needed here?

Comment (jiechenbo) [reply test/cppsuite/configs/search_near_01_stress.txt:23]: Yep! thats correct, key size is not utilised in search near 01. For each run, we randomise the search key size between 1 - 3, and use that size throughout the test.

File: test/cppsuite/configs/search_near_02_stress.txt [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        op_rate=10ms,
+        thread_count=10
+    ),
+    read_config=
+    (
+        # The key size parameter is used to generate the prefix 
+        # used in search_near. Use a smaller size to increase
+        # the chance to match an existing key.
```
Comment (etienneptl) [inline test/cppsuite/configs/search_near_02_stress.txt:29]: typo: chance*s*

**Review by etienneptl — APPROVED** [review]
LGTM, one small typo.

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Pull request merged**